src/sort/simple-sort-directive.js

```diff
@@ -0,0 +1,227 @@
+              compValue = item1.count - item2.count;
+          } else if ($scope.sortConfig.currentField.id === 'description') {
+            compValue = item1.description.localeCompare(item2.description);
+          }
+
+          if (!$scope.sortConfig.isAscending) {
+            compValue = compValue * -1;
+          }
+
+          return compValue;
+        };
+
+        var sortChange = function (sortId, isAscending) {
+          $scope.items.sort(compareFn);
+        };
+
+        $scope.sortConfig = {
+          fields: [
+            {
+              id: 'name',
+              title:  'Name',
+              sortType: 'alpha'
+            },
+            {
+              id: 'count',
+              title:  'Count',
+              sortType: 'numeric'
+            },
+            {
+              id: 'description',
+              title:  'Description',
+              sortType: 'alpha'
+            }
+          ],
+          onSortChange: sortChange
+        };
+      }
+    ]);
+  </file>
+</example>
+ */
+angular.module('patternfly.sort').directive('pfSimpleSort',
+  function () {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        config: '='
+      },
+      templateUrl: 'sort/simple-sort.html',
+      controller: function ($scope) {
+        $scope.setupConfig = function () {
+          var updated = false;
+
+          if ($scope.config.fields === undefined) {
+            $scope.config.fields = [];
+          }
+
+          if ($scope.config.fields.length > 0) {
+            if ($scope.config.currentField === undefined) {
+              $scope.config.currentField = $scope.config.fields[0];
+              updated = true;
+            }
+            if ($scope.config.isAscending === undefined) {
+              $scope.config.isAscending = true;
+              updated = true;
+            }
+          }
+
+          if (updated === true && $scope.config.onSortChange) {
+            $scope.config.onSortChange($scope.config.currentField, $scope.config.isAscending);
+          }
+        };
+
+        $scope.selectField = function (field) {
+          $scope.config.currentField = field;
+
+          if ($scope.config.onSortChange) {
+            $scope.config.onSortChange($scope.config.currentField, $scope.config.isAscending);
+          }
+        };
+
+        $scope.changeDirection = function () {
+          $scope.config.isAscending = !$scope.config.isAscending;
+
+          if ($scope.config.onSortChange) {
+            $scope.config.onSortChange($scope.config.currentField, $scope.config.currentDirection);
+          }
+        };
+
+        $scope.getSortIconClass = function () {
+          var iconClass;
+
+          if ($scope.config.currentField.sortType === 'alpha') {
+            if ($scope.config.isAscending) {
+              iconClass = 'fa fa-sort-alpha-asc';
+            } else {
+              iconClass = 'fa fa-sort-alpha-desc';
```

Comment:
There are 4 variations all which would check almost the same things. I thought it clearer to provide a function for this logic.